packages/smart-contracts/src/contracts/ERC20SwapToPay.sol

```diff
@@ -82,15 +107,33 @@ contract ERC20SwapToPay is Ownable {
     IERC20 spentToken = IERC20(_path[0]);
     IERC20 requestedToken = IERC20(_path[_path.length-1]);
 
+    
     uint256 requestedTotalAmount = _amount + _feeAmount;
 
-    require(spentToken.allowance(msg.sender, address(this)) > _amountInMax,
-                                 "Not sufficient allowance for swap to pay.");
+    // make sure the allowance includes the swapFee.
+    require(
+      spentToken.allowance(
+        msg.sender, 
+        address(this)) > (_amountInMax + _calculateSwapFee(requestedTotalAmount)
```

Comment:
It's probably wise to call `_calculateSwapFee` only once and to use the constant where needed.